go/pkg/cas/client.go

```diff
@@ -232,8 +240,8 @@ func (c *Client) init() {
 	c.semBatchUpdateBlobs = semaphore.NewWeighted(int64(c.BatchUpdateBlobs.Concurrency))
 
 	c.semFileIO = semaphore.NewWeighted(int64(c.FSConcurrency))
-	c.fileIOBufs.New = func() interface{} {
-		return make([]byte, c.FileIOSize)
+	c.fileBufReaders.New = func() interface{} {
+		return bufio.NewReaderSize(emptyReader, int(c.FileIOSize))
```

Comment:
I think this can be `NewReaderSize(nil, ...)`?

If so, add comments explaining this is meant to always be reset with an actual reader.

Reply:
it happens to work (based on my reading of the code), but it is documented. I am slightly hesitant to rely on undocumented behavior, but I can change it to nil if yo prefer.

Reply:
emptyReader sgtm. I agree it seems better to not accidentally crash an user because of internal/golang changes handling nils.